.github/workflows/build-test-windows.yml

```diff
@@ -53,4 +53,13 @@ jobs:
     - name: Test MSVC x64
       shell: pwsh
       run: |
-        & .\build-msvc-x64\tests\nvapi64-tests.exe [@unit-tests]
\ No newline at end of file
+        & .\build-msvc-x64\tests\nvapi64-tests.exe [@unit-tests]
+
+    - name: Upload artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: dxvk-nvapi-msvc
+        path: |
+          .\build-msvc-x86\src\nvapi.dll
+          .\build-msvc-x64\src\nvapi64.dll
+          .\build-msvc-x64\tests\nvapi64-tests.exe
```

Comment:
Okay, a few points of interest about this:

1. I'm not sure if there's any point in including the test binary. Feel free to prove me wrong though.
1. Specifying multiple paths like this will preserve the directory structure which is probably uninteresting for potential consumers of this artifact. A glob like `./build-msvc-*/src/nvapi*.dll` will catch the DLLs and put them at the top level of the archive. (As a side note, I dislike Windows-style path( separator)s but both are documented to be supported so it doesn't matter.)
1. On the other hand, there _could_ be some value in persisting debug symbol databases, but maybe as a separate artifact that catches `./build-msvc-*/src/nvapi*.pdb` as something like `dxvk-nvapi-msvc-dbgsym`.

Reply:
Actually, `./build-msvc-*/src/*.dll` glob would be a bit more future-proof because we won't have to update it after Liam's nvofapi PR.

Reply:
good points, thanks! Last push should better (if it works like I think it does).

Reply:
> No files were found with the provided path: .\build-msvc-x*\[src][tests]\*.[dll][exe][pdb]. No artifacts will be uploaded.

Yeah, `[…]` doesn't work like that, we would need something like bash's `{foo,bar}` but that doesn't seem to be supported.

Reply:
oh thank you github for making this easy ;). I guess in that case I would prefer to just keep the directory structure. 
So like this:

```
.\build-msvc-*\src\*.dll
.\build-msvc-*\src\*.pdb
.\build-msvc-*\tests\*.exe
.\build-msvc-*\tests\*.pdb
```

Reply:
Hmm, weird, it still didn't catch the PDBs. I wonder if they are even generated with current configuration :thinking:

Welp, I'd leave this for another day to figure out.

Reply:
About you original points:
1. I consider this artifact for now for debugging purposes, having the tests exe makes it easy to just run anything against the build dll's, at least that was the first thing I did after adding the artifacts upload.
2. See earlier discussion. I'm fine to keep the directory structure considering the github limitations. If we want to distribute this more actively, we would need to include readme/licence and adjust the layout anyway to match the readme.
3. I don't mind slashes/backslashes here, since is a Windows runner, I went for consistency with the rest of the yml.

Reply:
> I went for consistency with the rest of the yml.

It's already quite inconsistent if we consider `msbuild -m build-msvc-x64/dxvk-nvapi.sln` :see_no_evil:

Reply:
> > I went for consistency with the rest of the yml.
> 
> It's already quite inconsistent if we consider `msbuild -m build-msvc-x64/dxvk-nvapi.sln` 🙈

Right.. ;)  I've adjusted that while being at it.

Reply:
mmh, no pdb's in the zip..

Reply:
I guess because of `--buildtype release`? It has been a while since I used msvc..